pandas/core/groupby.py

```diff
@@ -2201,36 +2201,45 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                     raise AssertionError('Level %s not in index' % str(level))
                 level = index.names.index(level)
 
-            inds = index.labels[level]
-            level_index = index.levels[level]
-
             if self.name is None:
                 self.name = index.names[level]
 
-            # XXX complete hack
+            if isinstance(index, MultiIndex):
+                inds = index.labels[level]
```

Comment:
i think this should be a private method in an Index instead (and overridden in MultiIndex)